.github/workflows/release.yml

```diff
@@ -1,26 +1,116 @@
 name: Release
 
 on:
-  push:
-    tags:
-      - '[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      releaseType:
+        type: choice
+        options:
+          - minor
+          - patch
+        default: minor
+        required: true
+        description: 'minor: v0.X.0, patch: v0.0.X'
+      debug:
+        type: boolean
+        default: true
+        description: 'executes the workflow in debug mode (skip publishing tag, release and docker image)'
 
 jobs:
-  release:
-    name: Create Release
-    if: github.event.base_ref == 'refs/heads/master'
+  check-permission:
+    name: Check permission
+    if: contains(github.ref, 'refs/heads/master')
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
     steps:
-      - name: Get Version
-        id: get_version
+      - name: Check user permission
+        uses: actions-cool/check-user-permission@v2.2.0
+        id: check
+        with:
+          require: 'write'
+    outputs:
+      hasWritePermission: ${{ steps.check.outputs.require-result }}
+
+  publish-tag:
+    name: Publish tag
+    needs: check-permission
+    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout prebid cache
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Create & publish tag
+        id: release
+        run: |
+          currentTag=$(git describe --abbrev=0 --tags)
+          echo "current release tag ${currentTag}"
```

Comment:
I'm not sure if I'm being overcautious here. So far we've been creating the release tag manually and there may be a scenario in the future where we may create a tag manually again (a major release, maybe). Should we add a check in case we make a mistake on a tag? The rest of the script asumes, with good reason, that the previous tag always has the following format: `oneOrMoreDigits.oneOrMoreDigits.oneOrMoreDigits`

If, for some reason, we manually (and probably by mistake) create and push the following tag: `0.12y.0`, should we check and exit before running the `awk` command? With this hipothetical output, `awk` will compute a minor of `0.13.0` which is a tag that already exists:
```
╰─➤  echo "$(echo "0.12y.0" | awk -F. '{OFS="."; $2+=1; $3=0; print $0}')"
0.13.0
```
I'm not an `awk` wizard, but, if there isn't a way to do the check there, `grep` does the trick :
```
        run: |
          currentTag=$(git describe --abbrev=0 --tags)
 +        echo "${currentTag}" | grep -q "[0-9]\+\.[0-9]\+\.[0-9]\+"
 +        if [ $? -ne 0 ]; then
 +          echo "Current tag format won't let us compute the new tag name. Would you like to manually enter the tag instead?"
 +          # 
 +          # ...
 +          #
 +        fi
          echo "current release tag ${currentTag}"

          nextTag=''
```
Thoughts?

Reply:
make sense to add more validation to catch manually pushed tag which is not in expected format

https://github.com/prebid/prebid-cache/pull/135/commits/cf45cd47c5a68ba25d94b1d3f2f86ebe204c6c76 makes the changes to validate tag. Script aborts the execution with exit 1 status for tags with unexpected format